content/docs/learning-guide/tutorials/custom-tools/shape-example-cpp.md

```diff
@@ -0,0 +1,383 @@
+
+   - Format the name and suffix, if any.
+
+    - Call `EditorEntityAPIBus::Event(...)` and dispatch `EditorEntityAPIRequests::SetName` to set the name of `newEntityId` to `entityName`. 
+
+4. Add a Shape component to the entity. Call `EditorComponentAPIBus::Broadcast(...)` and dispatch `EditorComponentAPIRequests::AddComponentsOfType`, which adds the components from the provided list to `newEntityId`.
+
+```cpp
+    void ShapeExampleWidget::CreateEntityWithShapeComponent(const AZ::TypeId& typeId)
+    {
+        // 1
+        using namespace AzToolsFramework;
+
+        // 2
+        AZ::EntityId newEntityId;
+        EditorRequestBus::BroadcastResult(newEntityId, &EditorRequests::CreateNewEntity, AZ::EntityId());
+
+        // 3
+        if (!m_nameInput->text().isEmpty())
+        {
+            QString entityName = m_nameInput->text();
+
+            if (m_addShapeNameSuffix->isChecked())
+            {
+                AZStd::vector<AZStd::string> componentNames;
+                EditorComponentAPIBus::BroadcastResult(
+                    componentNames,
+                    &EditorComponentAPIRequests::FindComponentTypeNames,
+                    AZ::ComponentTypeList{ typeId }
+                );
+
+                AZ_Assert(componentNames.size() == 1, "Unable to find component name");
+
+                QString shapeName(componentNames[0].c_str());
+                shapeName.replace(" ", "");
+
+                entityName = QString("%1_%2").arg(entityName).arg(shapeName);
+            }
+            
+            EditorEntityAPIBus::Event(newEntityId, &EditorEntityAPIRequests::SetName, entityName.toUtf8().constData());
+        }
+
+        // 4
+        EditorComponentAPIBus::Broadcast(&EditorComponentAPIRequests::AddComponentsOfType, newEntityId, AZ::ComponentTypeList{ typeId });
+    }
+```
+
+
+## Icon
+
+An icon is an image file that's used to represent your tool in the O3DE Editor. You will store the icon using the Qt Resource System and load it from your Gem module. We recommend that your icon adheres to the guidelines in [UI development best practices](https://o3de.org/docs/tools-ui/uidev-component-development-guidelines/#ui-development-best-practices).
+
+### Add an icon
+
+1. Add an image file to the `Code/Source` directory to use as your icon.
+   
+2. Add your icon to your Gem's resources by updating `ShapeExample.qrc` with your new icon's file name. 
+
+3. Register your Gem's resources in Qt by adding the following code in `EditorModule.cpp`. 
+
+    - Define the function `InitShapeExampleResource()` and call `Q_INIT_RESOURCE(...)` to register the Qt resrouces listed in `ShapeExample.qrc`.
+
+        ```cpp
+        void InitShapeExampleResources()
+        {
+            Q_INIT_RESOURCE(ShapeExample);
+        }
+        ```
+
+    - Call `InitShapeExampleResource()` in the class constructor.
+
```

Comment:
Add a new call out heading / section here. Similar to what "Icon" is styled as. Maybe you can call it "Download the Sample file"

And have an animated gif of the sample being used in the editor (5-7s loop?)